_pytest/python.py

```diff
@@ -502,6 +502,8 @@ def _get_xunit_func(obj, name):
 class Class(PyCollector):
     """ Collector for test methods. """
     def collect(self):
+        if not getattr(self.obj, "__test__", True):
```

Comment:
`pytest.compat` has a `safe_getattr` which returns a default when exceptions happen - shouldn't this be used here? (see #214)